src/main/kotlin/no/nav/syfo/integration/kafka/journalpost/JournalpostHendelseConsumer.kt

```diff
@@ -0,0 +1,116 @@
+package no.nav.syfo.integration.kafka.journalpost
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import no.nav.helse.arbeidsgiver.bakgrunnsjobb.Bakgrunnsjobb
+import no.nav.helse.arbeidsgiver.bakgrunnsjobb.BakgrunnsjobbRepository
+import no.nav.helse.arbeidsgiver.kubernetes.LivenessComponent
+import no.nav.helse.arbeidsgiver.kubernetes.ReadynessComponent
+import no.nav.syfo.kafkamottak.InngaaendeJournalpostDTO
+import no.nav.syfo.prosesser.JoarkInntektsmeldingHendelseProsessor
+import no.nav.syfo.repository.DuplikatRepository
+import org.apache.avro.generic.GenericRecord
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.slf4j.LoggerFactory
+import java.time.Duration.ofMillis
+import java.time.LocalDateTime
+
+class JournalpostHendelseConsumer(
+    props: Map<String, Any>,
+    topicName: String,
+    private val duplikatRepository: DuplikatRepository,
+    private val bakgrunnsjobbRepo: BakgrunnsjobbRepository,
+    private val om: ObjectMapper
+) : ReadynessComponent, LivenessComponent {
+
+    private val log = LoggerFactory.getLogger(JournalpostHendelseConsumer::class.java)
+    private val consumer: KafkaConsumer<String, GenericRecord> = KafkaConsumer(props)
+    private var ready = false
+    private var error = false
+
+    init {
+        log.info("Lytter på topic $topicName")
+        consumer.subscribe(listOf(topicName))
+    }
+
+    fun setIsReady(ready: Boolean) {
+        this.ready = ready
+    }
+
+    fun setIsError(isError: Boolean) {
+        this.error = isError
+    }
+
+    fun start() {
+        var totalCount = 0L
+        log.info("Starter...")
+        consumer.use {
+            setIsReady(true)
+            while (!error) {
+                consumer
+                    .poll(ofMillis(1000))
+                    .fold(totalCount) { accumulator, record ->
+                        val newCount = accumulator + 1
+                        try {
+                            try {
+                                processHendelse(mapJournalpostHendelse(record.value()))
+                            } catch (ex: Throwable) {
+                                throw IllegalArgumentException("Klarte ikke lese hendelse med offset ${record.offset()}!", ex)
+                            }
+                            consumer.commitSync()
+                        } catch (e: Throwable) {
+                            log.error("Klarte ikke behandle hendelse. Stopper lytting!", e)
+                            setIsError(true)
+                        }
+                        newCount
+                    }
+            }
+        }
+    }
+
+    fun processHendelse(journalpostDTO: InngaaendeJournalpostDTO): Int {
```

Comment:
Hvorfor returnerer denne int?

Reply:
Det er for å kunne gjøre koden testbar på det viktiste i klassen: utfallet på av behandlingen av journalposten.

Reply:
Jeg er ikke spesielt fan av at tester påvirker koden. Her kan det også være fort gjort å endre på faktisk oppførsel til funksjonen, uten at testen plukker det opp.